internal/pkg/service/buffer/statistics/apinode_test.go

```diff
@@ -0,0 +1,160 @@
+package statistics_test
+
+import (
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/benbjohnson/clock"
+	"github.com/keboola/go-utils/pkg/wildcards"
+
+	. "github.com/keboola/keboola-as-code/internal/pkg/service/buffer/statistics"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/buffer/store/key"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/dependencies"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/errors"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/etcdhelper"
+)
+
+func TestStatsManager(t *testing.T) {
+	t.Parallel()
+
+	clk := clock.NewMock()
+	d := dependencies.NewMockedDeps(t, dependencies.WithClock(clk))
+	client := d.EtcdClient()
+	node := NewAPINode(d)
+
+	sliceKey := key.NewSliceStatsKey(123, "my-receiver", "my-export", clk.Now(), clk.Now(), "my-node")
+	clk.Add(time.Hour)
+
+	// no notify -> wait 1 second -> no sync
+	clk.Add(SyncInterval)
+	etcdhelper.AssertKVs(t, client, "")
+
+	// notify -> wait 1 second -> sync
+	node.Notify(sliceKey, 1000)
+	etcdhelper.ExpectModification(t, client, func() {
+		clk.Add(SyncInterval)
+	})
+	etcdhelper.AssertKVs(t, client, `
+<<<<<
+stats/received/123/my-receiver/my-export/1970-01-01T00:00:00.000Z/1970-01-01T00:00:00.000Z/my-node
+-----
+{
+  "projectId": 123,
+  "receiverId": "my-receiver",
+  "exportId": "my-export",
+  "fileId": "1970-01-01T00:00:00Z",
+  "sliceId": "1970-01-01T00:00:00Z",
+  "nodeId": "my-node",
+  "count": 1,
+  "size": 1000,
+  "lastReceivedAt": "1970-01-01T01:00:01Z"
+}
+>>>>>
+`)
```

Comment:
Da sa to otestovat az po hodnoty v etcd.

Mozno mi na to povies, ze to uz nie je unit test, ale skor nejaky integracny test.

Cielom je otestovat to co najlepsie, ... a tu nam nestaci nejaky API E2E, musime ten test napisat rucne.
Teoreticky by mohol zostat povodny test a tento by sa k nemu pridal, ... ale kedze by ten unit test testoval to iste, iba menej E2E, tak mi to nedavalo zmysel.